yt_dlp/extractor/sheeta.py

```diff
@@ -0,0 +1,836 @@
+                fanclub_list_json, ('data', 'content_providers', 'id', {int_or_none}), get_all=False):
+            return fanclub_id
+        raise ExtractorError(f'Channel {channel_id} does not exist', expected=True)
+
+    def _extract_base_info(self, channel_id):
+        site_settings = self._download_json(
+            f'https://{self._DOMAIN}/site/settings.json', None,
+            note='Fetching site settings', errnote='Unable to fetch site settings')
+        self.write_debug(f'site_settings = {site_settings!r}')
+
+        self._API_BASE_URL = site_settings['api_base_url']
+        self._FANCLUB_GROUP_ID = site_settings['fanclub_group_id']
+        self._FANCLUB_SITE_ID_AUTH = site_settings['fanclub_site_id']
+
+        if channel_id:
+            self._FANCLUB_SITE_ID_INFO = self._find_fanclub_site_id(channel_id)
+        else:
+            self._FANCLUB_SITE_ID_INFO = self._FANCLUB_SITE_ID_AUTH
+
+    @property
+    def _auth_info(self):
+        if not self._AUTH_INFO.get(self._DOMAIN):
+            self._AUTH_INFO[self._DOMAIN] = {}
+        return self._AUTH_INFO.get(self._DOMAIN)
+
+    @_auth_info.setter
+    def _auth_info(self, value):
+        if not self._AUTH_INFO.get(self._DOMAIN):
+            self._AUTH_INFO[self._DOMAIN] = {}
+        self._AUTH_INFO[self._DOMAIN].update(value)
+
+    @property
+    def _channel_base_info(self):
+        return traverse_obj(self._call_api(
+            f'fanclub_sites/{self._FANCLUB_SITE_ID_INFO}/page_base_info', f'fanclub_sites/{self._FANCLUB_SITE_ID_INFO}',
+            note='Fetching channel base info', errnote='Unable to fetch channel base info', fatal=False,
+        ), ('data', 'fanclub_site', {dict})) or {}
+
+    @property
+    def _channel_user_info(self):
+        return traverse_obj(self._call_api(
+            f'fanclub_sites/{self._FANCLUB_SITE_ID_INFO}/user_info', f'fanclub_sites/{self._FANCLUB_SITE_ID_INFO}',
+            note='Fetching channel user info', errnote='Unable to fetch channel user info', fatal=False,
+            data=json.dumps('null').encode(),
+        ), ('data', 'fanclub_site', {dict})) or {}
+
+    def _extract_channel_info(self, channel_id):
+        if channel_id:
+            full_channel_id = f'{self._DOMAIN}/{channel_id}'
+            channel_url = f'https://{self._DOMAIN}/{channel_id}'
+        else:
+            full_channel_id = self._DOMAIN
+            channel_url = f'https://{self._DOMAIN}'
+
+        return {
+            'channel': self._channel_base_info.get('fanclub_site_name'),
+            'channel_id': full_channel_id,
+            'channel_url': channel_url,
+            'age_limit': traverse_obj(self._channel_user_info, (
+                'content_provider', 'age_limit', {int_or_none})),
+        }
+
+    def _extract_player_page(self, url):
+        self._DOMAIN, channel_id, content_code = re.match(
+            r'https?://(?P<domain>[\w.-]+)(/(?P<channel>[\w.-]+))?/(?:live|video|audio)/(?P<code>sm\w+)', url
```

Comment:
Any reason why you have been using `code` instead of `id` for the ID of videos in the NNC regexes? This prevents `InfoExtractor._match_id` and thus archive lookups from working.

Reply:
Hi.

> This prevents InfoExtractor._match_id and thus archive lookups from working.

I know the `_match_id` function will not work without the "id" group. What is "archive lookups"? Can you please elaborate?

Reply:
Finished videos have entries in the file specified by `--download-archive`, so they are skipped in subsequent downloads e.g. when running yt-dlp for a bunch of channels in a timer service. Fixing this is one of the patches I maintain for myself. Another NNC specific patch I have is to get rid of the `[download] Downloading item %s of %s` message, even if it's skipped due to having an archive entry, although this might affect more extractors that I have yet to use.

Reply:
Thanks for your reply.

> Finished videos have entries in the file specified by `--download-archive`, so they are skipped in subsequent downloads e.g. when running yt-dlp for a bunch of channels in a timer service.

Well, I don't know that the "id" group in the regex will affect the function of `--download-archive`. I also don't see any reason not to use "id" here. I may have cleared my code incorrectly. I will fix it.

> Another NNC specific patch I have is to get rid of the `[download] Downloading item %s of %s` message, ...

Hmm, did this message affect the functionality of this extractor?

Reply:
> Hmm, did this message affect the functionality of this extractor?

No, but other extractors (at least `youtube` and `twitchtv`) don't have this behavior and it also clutters my journald entries. You can leave it as is, I already have other modifications to make things less verbose, just thought I'd mention it since I'm already commenting on the refactor(?) of this extractor.

Reply:
> No, but other extractors (at least `youtube` and `twitchtv`) don't have this behavior and it also clutters my journald entries.

OK, maybe you have to hold a forked yt-dlp and preserve your changes in the core.

> just thought I'd mention it since I'm already commenting on the refactor(?) of this extractor.

Thanks for your information.

This refactor is not yet complete and there is no ETA because I'm busy with other things. If you'd like, you can open PRs in my fork and point to the correct branch. We can work together to finish the new code.